src/MSALWrapper/PCACache.cs

```diff
@@ -88,6 +100,153 @@ public void SetupTokenCache(ITokenCache userTokenCache, IList<Exception> errors)
             }
         }
 
+        /// <summary>
+        /// Sets up a plain text cache fallback for headless Linux environments.
+        /// </summary>
+        /// <param name="userTokenCache">An <see cref="ITokenCache"/> to use.</param>
+        /// <param name="errors">The errors list to append error encountered to.</param>
+        private void SetupPlainTextCache(ITokenCache userTokenCache, IList<Exception> errors)
+        {
+            try
+            {
+                var homeDir = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+                var cacheDir = Path.Combine(homeDir, PlainTextCacheDir);
+                var cacheFilePath = Path.Combine(cacheDir, PlainTextCacheFileName);
+
+                // Create directory if it doesn't exist
+                if (!Directory.Exists(cacheDir))
+                {
+                    Directory.CreateDirectory(cacheDir);
+                    // Set directory permissions to user only (700)
+                    SetDirectoryPermissions(cacheDir);
+                }
+
+                // Create or ensure cache file exists with proper permissions
+                if (!File.Exists(cacheFilePath))
+                {
+                    File.WriteAllText(cacheFilePath, "{}");
+                    SetFilePermissions(cacheFilePath);
+                }
+                else
+                {
+                    // Ensure existing file has proper permissions
+                    SetFilePermissions(cacheFilePath);
+                }
+
+                var storageProperties = new StorageCreationPropertiesBuilder(PlainTextCacheFileName, cacheDir)
+                    .WithUnprotectedFile()
+                    .Build();
+
+                MsalCacheHelper cacher = MsalCacheHelper.CreateAsync(storageProperties).Result;
+                cacher.RegisterCache(userTokenCache);
+
+                this.logger.LogInformation($"Plain text cache fallback configured at: {cacheFilePath}");
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogWarning($"Plain text cache fallback failed: {ex.Message}");
+                errors.Add(ex);
+            }
+        }
+
+        /// <summary>
+        /// Checks if the current platform is Linux.
+        /// </summary>
+        /// <returns>True if running on Linux, false otherwise.</returns>
+        private static bool IsLinux()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+        }
+
+        /// <summary>
+        /// Checks if the current Linux environment is headless (no display server).
+        /// </summary>
+        /// <returns>True if headless Linux environment, false otherwise.</returns>
+        private static bool IsHeadlessLinux()
+        {
+            // Check if DISPLAY environment variable is not set or empty

```

Comment:
I don't yet know for certain if this is true, but I'm not sure that a Wayland environment is _required_ to have `$DISPLAY` set and might actually just have `$WAYLAND_DISPLAY` set. As written, I wonder if this would cause a native Wayland environment to prematurely be detected as headless. Was this tested in a Wayland environment?